packages/@webex/internal-plugin-metrics/src/call-diagnostic/call-diagnostic-metrics-latencies.ts

```diff
@@ -0,0 +1,290 @@
+    // TODO: not clear
+    return undefined;
+  }
+
+  /**
+   * Local SDP Generated Remote SDP REceived
+   * @returns - latency
+   */
+  public getLocalSDPGenRemoteSDPRecv() {
+    return this.getDiffBetweenTimestamps(
+      'client.media-engine.local-sdp-generated',
+      'client.media-engine.remote-sdp-received'
+    );
+  }
+
+  /**
+   * ICE Setup Time
+   * @returns - latency
+   */
+  public getICESetupTime() {
+    return this.getDiffBetweenTimestamps('client.ice.start', 'client.ice.end');
+  }
+
+  /**
+   * Audio ICE time
+   * @returns - latency
+   */
+  public getAudioICESetupTime() {
+    return this.getDiffBetweenTimestamps('client.ice.start', 'client.ice.end');
+  }
+
+  /**
+   * Video ICE Time
+   * @returns - latency
+   */
+  public getVideoICESetupTime() {
+    return this.getDiffBetweenTimestamps('client.ice.start', 'client.ice.end');
+  }
+
+  /**
+   * Share ICE Time
+   * @returns - latency
+   */
+  public getShareICESetupTime() {
+    return this.getDiffBetweenTimestamps('client.ice.start', 'client.ice.end');
+  }
+
+  /**
+   * Stay Lobby Time
+   * @returns - latency
+   */
+  public getStayLobbyTime() {
+    return this.getDiffBetweenTimestamps(
+      'client.locus.join.response',
+      'host.meeting.participant.admitted'
+    );
+  }
+
+  /**
+   * Page JMT
+   * @returns - latency
+   */
+  public getPageJMT() {
+    return this.latencyTimestamps.get('client.pageJMT.received') || undefined;
+  }
+
+  /**
+   * Click To Interstitial
+   * @returns - latency
+   */
+  public getClickToInterstitial() {
+    return this.getDiffBetweenTimestamps(
+      'client.meeting.click.joinbutton',
+      'client.meeting.interstitial-window.showed'
+    );
+  }
+
+  /**
+   * Interstitial To Join Ok
+   * @returns - latency
+   */
+  public getInterstitialToJoinOK() {
+    return this.getDiffBetweenTimestamps(
+      'client.meeting.click.joinbutton',
+      'client.locus.join.response'
+    );
+  }
+
+  /**
+   * Interstitial To Media Ok
+   * @returns - latency
+   */
+  public getInterstitialToMediaOK() {
+    return this.getDiffBetweenTimestamps(
+      'client.meeting.click.joinbutton',
+      'sdk.media-flow.started'
+    );
+  }
+
+  /**
+   * Total JMT
+   * @returns - latency
+   */
+  public getTotalJMT() {
+    const clickToInterstitial = this.getClickToInterstitial();
+    const interstitialToJoinOk = this.getInterstitialToJoinOK();
+
+    if (clickToInterstitial && interstitialToJoinOk) {
+      return clickToInterstitial + interstitialToJoinOk;
+    }
+
+    return undefined;
+  }
+
+  /**
+   * Join Conf JMT
+   * @returns - latency
+   */
+  public getJoinConfJMT() {
+    const joinReqResp = this.getJoinReqResp();
+    const ICESetupTime = this.getICESetupTime();
+
+    if (joinReqResp && ICESetupTime) {
+      return joinReqResp + ICESetupTime;
+    }
+
+    return undefined;
+  }
+
+  /**
+   * Total Media JMT
+   * @returns - latency
+   */
+  public getTotalMediaJMT() {
```

Comment:
this one is same as getClientJMT + getClickToInterstitial. would it be better to implement it like that?

Reply:
I would like to keep the implementation as close to the formula provided by CA